library/init/category/functor.lean

```diff
@@ -22,9 +19,4 @@ infixr ` <$ `:100  := has_map.map_const
 λ a b, b <$ a
 infixr ` $> `:100  := has_map.map_const_rev
 
-class functor (f : Type u → Type v) extends has_map f : Type (max (u+1) v) :=
-(map_const_eq : ∀ {α β : Type u}, @map_const α β = map ∘ const β . control_laws_tac)
--- `functor` is indeed a categorical functor
-(id_map       : Π {α : Type u} (x : f α), id <$> x = x)
-(map_comp     : Π {α β γ : Type u} (g : α → β) (h : β → γ) (x : f α), (h ∘ g) <$> x = h <$> g <$> x)
-end
+class functor (f : Type u → Type v) extends has_map f
```

Comment:
It seems we don't need `has_map` anymore. We can keep `functor` only.
I believe we had both because `has_map` was the lawless `functor`.

Reply:
I assumed it was a "functor without pure", like `has_bind` was introduced for "monads without return"

Reply:
I didn't understand your answer. In this PR, `functor` and `has_map` are morally identical and have the same fields.
They would have the same fields if we were still flattening structures.

Reply:
Sorry, I got my monad hierarchy mixed up. I'll remove it.